test/geometry_visual_test.cpp

```diff
@@ -0,0 +1,85 @@
+#include <mapnik/geometry_is_valid.hpp>
+#include <mapnik/geometry_is_simple.hpp>
+
+#include "catch.hpp"
+#include "encoding_util.hpp"
+#include "test_utils.hpp"
+#include "vector_tile_strategy.hpp"
+#include "vector_tile_processor.hpp"
+#include "vector_tile_backend_pbf.hpp"
+#include "vector_tile_projection.hpp"
+#include "vector_tile_geometry_decoder.hpp"
+
+mapnik::geometry::geometry<double> clip_geometry(mapnik::box2d<double> bbox, std::string file)
+{
+    typedef mapnik::vector_tile_impl::backend_pbf backend_type;
+    typedef mapnik::vector_tile_impl::processor<backend_type> renderer_type;
+    typedef vector_tile::Tile tile_type;
+    std::shared_ptr<mapnik::memory_datasource> ds = testing::build_geojson_ds(file);
+    tile_type tile;
+    backend_type backend(tile,1000);
+    unsigned tile_size = 256;
+    // ds->set_envelope(bbox);
+
+    mapnik::Map map(tile_size,tile_size,"+init=epsg:3857");
+    mapnik::layer lyr("layer","+init=epsg:4326");
+    lyr.set_datasource(ds);
+    map.add_layer(lyr);
+    map.zoom_to_box(bbox);
+    mapnik::request m_req(tile_size,tile_size,bbox);
+    renderer_type ren(backend,map,m_req);
+    ren.apply();
+    std::string buffer;
+    tile.SerializeToString(&buffer);
+    if (buffer.size() > 0 && tile.layers_size() > 0 && tile.layers_size() != false)
+    {
+        vector_tile::Tile_Layer const& layer = tile.layers(0);
+        if (layer.features_size() != false)
+        {
+            vector_tile::Tile_Feature const& f = layer.features(0);
+            unsigned z = 0;
+            unsigned x = 0;
+            unsigned y = 0;
+            double resolution = mapnik::EARTH_CIRCUMFERENCE/(1 << z);
+            double tile_x = -0.5 * mapnik::EARTH_CIRCUMFERENCE + x * resolution;
+            double tile_y =  0.5 * mapnik::EARTH_CIRCUMFERENCE - y * resolution;
+            double scale = (static_cast<double>(layer.extent()) / tile_size) * tile_size/resolution;
+            mapnik::vector_tile_impl::Geometry<double> geoms(f,tile_x, tile_y,scale,-1*scale);
+            mapnik::geometry::geometry<double> geom = mapnik::vector_tile_impl::decode_geometry<double>(geoms,f.type());
+
+            unsigned int n_err = 0;
+            mapnik::projection wgs84("+init=epsg:4326",true);
+            mapnik::projection merc("+init=epsg:3857",true);
+            mapnik::proj_transform prj_trans(merc,wgs84);
+            return mapnik::geometry::reproject_copy(geom,prj_trans,n_err);
+        }
+    }
+    return NULL;
+}
+
+TEST_CASE( "geometries", "should reproject, clip, and simplify")
+{
+
+    std::vector<std::string> geometries = mapnik::util::list_directory("./test/geometry-test-data/input");
+    for (std::string file: geometries)
+    {
+        std::shared_ptr<mapnik::memory_datasource> ds = testing::build_geojson_ds(file);
+        // mapnik::box2d<double> bbox = ds->envelope();
+        mapnik::box2d<double> bbox(-20037508.342789,-20037508.342789,20037508.342789,20037508.342789);
```

Comment:
@jakepruitt this is due to the reprojection that takes place on the map, you are assuming that your coordinates are in EPSG:4326 for the datasource then the maps extent is in EPSG:3857. Therefore, the extent of the datasource needs to be reprojected as well. Otherwise you are going to be zooming into a location with no data.